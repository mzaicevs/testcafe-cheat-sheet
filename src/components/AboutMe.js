import React from "react";
import { Card, ListGroup, CardDeck } from "react-bootstrap";
import LinkedInBadge from "./LinkedInBadge";
import CardLink from "../base_components/CardLink";

const WhatIsThisCardContent = () => (
  <Card.Body>
    <Card.Title>What is this?</Card.Title>
    <Card.Text>
      If you haven't guessed by now, this web app is designed as a Cheat Sheet
      to TestCafe. It's aim is to accelerate your usage of TestCafe with quick
      information. It's not designed to be a replacement for TestCafe's
      documentation, but rather a quick "what was that thing again?" answer
      book.
    </Card.Text>
    <Card.Text>
      Maybe the Jumbotron above wasn't as explicit as I had intended?
      <span role="img" aria-label="thinking-face">
        {" "}
        ðŸ¤”
      </span>
    </Card.Text>
  </Card.Body>
);

const WhoCreatedThisCardContent = () => (
  <Card.Body>
    <Card.Title>Who created this... thing?</Card.Title>
    <Card.Text>All blame can be pointed to:</Card.Text>
    <LinkedInBadge />
  </Card.Body>
);

const WhyWasThisCreatedCardContent = () => (
  <Card.Body>
    <Card.Title>Why would you create this?</Card.Title>
    <Card.Text>A number of reasons, really. The big drivers were:</Card.Text>
    <ul>
      <li>
        <Card.Text>
          I had started innovating at work with TestCafe on a front-end offering
          and constantly read and referred back to the TestCafe documentation,
          often for the next tidbit of information I needed. In fairness to the
          TestCafe maintainers, their documentation is nicely laid out and their
          search bar is super helpful. I just wanted to be selfish and have what
          I thought as the important information in a single page that I could
          refer back to with ease.
        </Card.Text>
      </li>
      <li>
        <Card.Text>
          <>
            I feel a sense of personal aimlessness if I'm not pumping out code
            or projects every so often. Partly inspired by
            <CardLink
              href="https://devhints.io/"
              text="devhints.io"
              withPreceedingSpace
            />
            ; I found an excuse to actually read the TestCafe docs more
            in-depth, experiment more with component libraries, and chase
            Lighthouse approval metrics. As it's mostly static information, I
            felt like it was low-hanging fruit I could pump out fairly quickly
            while learn and experiment that little bit more.
          </>
        </Card.Text>
      </li>
      <li>
        <Card.Text>
          What else would I do with myself?
          <span role="img" aria-label="zany-face">
            {" "}
            ðŸ¤ª
          </span>
        </Card.Text>
      </li>
    </ul>
  </Card.Body>
);

const WhatWasItMadeWithCardContent = () => (
  <Card.Body>
    <Card.Title>What is it made of?</Card.Title>
    <Card.Subtitle>Web App</Card.Subtitle>
    <ul>
      <li>
        <CardLink
          href="https://facebook.github.io/create-react-app/"
          text="create-react-app"
        />
        {` - React, and all the fanciness that goes with create-react-app!`}
      </li>
      <li>
        <CardLink
          href="https://react-bootstrap.netlify.com/"
          text={`react-bootstrap & bootstrap`}
        />
        {` - the lovely component library`}
      </li>
      <li>
        <CardLink
          href="https://github.com/nkbt/react-copy-to-clipboard"
          text="react-copy-to-cliboard"
        />
        {` - for all my copying needs`}
      </li>
      <li>
        <CardLink href="https://highlightjs.org/" text="highlight.js" />
        {` - code syntax highlighter`}
      </li>
    </ul>

    <Card.Subtitle>Build</Card.Subtitle>

    <ul>
      <li>
        <CardLink href="https://github.com/" text="Git" />
        {` - for the version control`}
      </li>
      <li>
        <CardLink href="https://travis-ci.org/" text="Travis" />
        {` - for the building on commit`}
      </li>
      <li>
        <CardLink href="https://pages.github.com/" text="Github Pages" />
        {` - for the deployment as a static web application`}
      </li>
    </ul>
    <Card.Subtitle>Test</Card.Subtitle>
    <ul>
      <li>
        <CardLink href="https://airbnb.io/enzyme/" text="Enzyme" />
      </li>
    </ul>
  </Card.Body>
);

const AboutMe = () => (
  <CardDeck>
    <Card>
      <ListGroup>
        <ListGroup.Item>
          <WhatIsThisCardContent />
        </ListGroup.Item>
        <ListGroup.Item>
          <WhoCreatedThisCardContent />
        </ListGroup.Item>
        <ListGroup.Item>
          <WhyWasThisCreatedCardContent />
        </ListGroup.Item>
        <ListGroup.Item>
          <WhatWasItMadeWithCardContent />
        </ListGroup.Item>
      </ListGroup>
    </Card>
  </CardDeck>
);

export default AboutMe;
