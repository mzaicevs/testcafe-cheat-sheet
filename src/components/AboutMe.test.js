import React from "react";
import { mount } from "enzyme";
import AboutMe from "./AboutMe";

describe("AboutMe", () => {
  it("matches snapshot", () => {
    const wrapper = mount(<AboutMe />);
    expect(wrapper).toMatchSnapshot();
  });
});
