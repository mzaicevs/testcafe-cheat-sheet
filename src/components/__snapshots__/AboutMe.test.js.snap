// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutMe matches snapshot 1`] = `
<AboutMe>
  <CardDeck>
    <div
      className="card-deck"
    >
      <Card
        as="div"
        body={false}
      >
        <div
          className="card"
        >
          <ForwardRef>
            <Uncontrolled(Bootstrap(ListGroup))
              innerRef={null}
            >
              <Bootstrap(ListGroup)
                onSelect={[Function]}
              >
                <ListGroup
                  as="div"
                  bsPrefix="list-group"
                  onSelect={[Function]}
                  variant={null}
                >
                  <ContextTransform(AbstractNav)
                    as="div"
                    className="list-group"
                    onSelect={[Function]}
                  >
                    <AbstractNav
                      as="div"
                      className="list-group"
                      onSelect={[Function]}
                    >
                      <div
                        className="list-group"
                        onKeyDown={[Function]}
                      >
                        <Bootstrap(ListGroupItem)>
                          <ListGroupItem
                            active={false}
                            bsPrefix="list-group-item"
                            disabled={false}
                            variant={null}
                          >
                            <ForwardRef
                              as="div"
                              className="list-group-item"
                              disabled={false}
                              eventKey={null}
                              onClick={[Function]}
                            >
                              <div
                                className="list-group-item"
                                data-rb-event-key={null}
                                disabled={false}
                                onClick={[Function]}
                              >
                                <WhatIsThisCardContent>
                                  <CardBody>
                                    <div
                                      className="card-body"
                                    >
                                      <CardTitle>
                                        <ForwardRef
                                          className="card-title"
                                        >
                                          <div
                                            className="card-title h5"
                                          >
                                            What is this?
                                          </div>
                                        </ForwardRef>
                                      </CardTitle>
                                      <CardText>
                                        <p
                                          className="card-text"
                                        >
                                          If you haven't guessed by now, this web app is designed as a Cheat Sheet to TestCafe. It's aim is to accelerate your usage of TestCafe with quick information. It's not designed to be a replacement for TestCafe's documentation, but rather a quick "what was that thing again?" answer book.
                                        </p>
                                      </CardText>
                                      <CardText>
                                        <p
                                          className="card-text"
                                        >
                                          Maybe the Jumbotron above wasn't as explicit as I had intended?
                                          <span
                                            aria-label="thinking-face"
                                            role="img"
                                          >
                                             
                                            ðŸ¤”
                                          </span>
                                        </p>
                                      </CardText>
                                    </div>
                                  </CardBody>
                                </WhatIsThisCardContent>
                              </div>
                            </ForwardRef>
                          </ListGroupItem>
                        </Bootstrap(ListGroupItem)>
                        <Bootstrap(ListGroupItem)>
                          <ListGroupItem
                            active={false}
                            bsPrefix="list-group-item"
                            disabled={false}
                            variant={null}
                          >
                            <ForwardRef
                              as="div"
                              className="list-group-item"
                              disabled={false}
                              eventKey={null}
                              onClick={[Function]}
                            >
                              <div
                                className="list-group-item"
                                data-rb-event-key={null}
                                disabled={false}
                                onClick={[Function]}
                              >
                                <WhoCreatedThisCardContent>
                                  <CardBody>
                                    <div
                                      className="card-body"
                                    >
                                      <CardTitle>
                                        <ForwardRef
                                          className="card-title"
                                        >
                                          <div
                                            className="card-title h5"
                                          >
                                            Who created this... thing?
                                          </div>
                                        </ForwardRef>
                                      </CardTitle>
                                      <CardText>
                                        <p
                                          className="card-text"
                                        >
                                          All blame can be pointed to:
                                        </p>
                                      </CardText>
                                      <LinkedInBadge>
                                        <div
                                          className="LI-profile-badge"
                                          data-locale="en_US"
                                          data-size="medium"
                                          data-theme="light"
                                          data-type="horizontal"
                                          data-vanity="byrne-greg"
                                          data-version="v1"
                                        >
                                          <a
                                            className="LI-simple-link"
                                            href="https://ie.linkedin.com/in/byrne-greg?trk=profile-badge"
                                          >
                                            Greg Byrne
                                          </a>
                                        </div>
                                      </LinkedInBadge>
                                    </div>
                                  </CardBody>
                                </WhoCreatedThisCardContent>
                              </div>
                            </ForwardRef>
                          </ListGroupItem>
                        </Bootstrap(ListGroupItem)>
                        <Bootstrap(ListGroupItem)>
                          <ListGroupItem
                            active={false}
                            bsPrefix="list-group-item"
                            disabled={false}
                            variant={null}
                          >
                            <ForwardRef
                              as="div"
                              className="list-group-item"
                              disabled={false}
                              eventKey={null}
                              onClick={[Function]}
                            >
                              <div
                                className="list-group-item"
                                data-rb-event-key={null}
                                disabled={false}
                                onClick={[Function]}
                              >
                                <WhyWasThisCreatedCardContent>
                                  <CardBody>
                                    <div
                                      className="card-body"
                                    >
                                      <CardTitle>
                                        <ForwardRef
                                          className="card-title"
                                        >
                                          <div
                                            className="card-title h5"
                                          >
                                            Why would you create this?
                                          </div>
                                        </ForwardRef>
                                      </CardTitle>
                                      <CardText>
                                        <p
                                          className="card-text"
                                        >
                                          A number of reasons, really. The big drivers were:
                                        </p>
                                      </CardText>
                                      <ul>
                                        <li>
                                          <CardText>
                                            <p
                                              className="card-text"
                                            >
                                              I had started innovating at work with TestCafe on a front-end offering and constantly read and referred back to the TestCafe documentation, often for the next tidbit of information I needed. In fairness to the TestCafe maintainers, their documentation is nicely laid out and their search bar is super helpful. I just wanted to be selfish and have what I thought as the important information in a single page that I could refer back to with ease.
                                            </p>
                                          </CardText>
                                        </li>
                                        <li>
                                          <CardText>
                                            <p
                                              className="card-text"
                                            >
                                              I feel a sense of personal aimlessness if I'm not pumping out code or projects every so often. Partly inspired by
                                              <CardLink
                                                href="https://devhints.io/"
                                                text="devhints.io"
                                                withPreceedingSpace={true}
                                              >
                                                <CardLink
                                                  href="https://devhints.io/"
                                                  rel="noopener noreferrer"
                                                  target="_blank"
                                                >
                                                  <a
                                                    className="card-link"
                                                    href="https://devhints.io/"
                                                    rel="noopener noreferrer"
                                                    target="_blank"
                                                  >
                                                     devhints.io
                                                  </a>
                                                </CardLink>
                                              </CardLink>
                                              ; I found an excuse to actually read the TestCafe docs more in-depth, experiment more with component libraries, and chase Lighthouse approval metrics. As it's mostly static information, I felt like it was low-hanging fruit I could pump out fairly quickly while learn and experiment that little bit more.
                                            </p>
                                          </CardText>
                                        </li>
                                        <li>
                                          <CardText>
                                            <p
                                              className="card-text"
                                            >
                                              What else would I do with myself?
                                              <span
                                                aria-label="zany-face"
                                                role="img"
                                              >
                                                 
                                                ðŸ¤ª
                                              </span>
                                            </p>
                                          </CardText>
                                        </li>
                                      </ul>
                                    </div>
                                  </CardBody>
                                </WhyWasThisCreatedCardContent>
                              </div>
                            </ForwardRef>
                          </ListGroupItem>
                        </Bootstrap(ListGroupItem)>
                        <Bootstrap(ListGroupItem)>
                          <ListGroupItem
                            active={false}
                            bsPrefix="list-group-item"
                            disabled={false}
                            variant={null}
                          >
                            <ForwardRef
                              as="div"
                              className="list-group-item"
                              disabled={false}
                              eventKey={null}
                              onClick={[Function]}
                            >
                              <div
                                className="list-group-item"
                                data-rb-event-key={null}
                                disabled={false}
                                onClick={[Function]}
                              >
                                <WhatWasItMadeWithCardContent>
                                  <CardBody>
                                    <div
                                      className="card-body"
                                    >
                                      <CardTitle>
                                        <ForwardRef
                                          className="card-title"
                                        >
                                          <div
                                            className="card-title h5"
                                          >
                                            What is it made of?
                                          </div>
                                        </ForwardRef>
                                      </CardTitle>
                                      <CardSubtitle>
                                        <ForwardRef
                                          className="card-subtitle"
                                        >
                                          <div
                                            className="card-subtitle h6"
                                          >
                                            Web App
                                          </div>
                                        </ForwardRef>
                                      </CardSubtitle>
                                      <ul>
                                        <li>
                                          <CardLink
                                            href="https://facebook.github.io/create-react-app/"
                                            text="create-react-app"
                                          >
                                            <CardLink
                                              href="https://facebook.github.io/create-react-app/"
                                              rel="noopener noreferrer"
                                              target="_blank"
                                            >
                                              <a
                                                className="card-link"
                                                href="https://facebook.github.io/create-react-app/"
                                                rel="noopener noreferrer"
                                                target="_blank"
                                              >
                                                create-react-app
                                              </a>
                                            </CardLink>
                                          </CardLink>
                                           - React, and all the fanciness that goes with create-react-app!
                                        </li>
                                        <li>
                                          <CardLink
                                            href="https://react-bootstrap.netlify.com/"
                                            text="react-bootstrap & bootstrap"
                                          >
                                            <CardLink
                                              href="https://react-bootstrap.netlify.com/"
                                              rel="noopener noreferrer"
                                              target="_blank"
                                            >
                                              <a
                                                className="card-link"
                                                href="https://react-bootstrap.netlify.com/"
                                                rel="noopener noreferrer"
                                                target="_blank"
                                              >
                                                react-bootstrap & bootstrap
                                              </a>
                                            </CardLink>
                                          </CardLink>
                                           - the lovely component library
                                        </li>
                                        <li>
                                          <CardLink
                                            href="https://github.com/nkbt/react-copy-to-clipboard"
                                            text="react-copy-to-cliboard"
                                          >
                                            <CardLink
                                              href="https://github.com/nkbt/react-copy-to-clipboard"
                                              rel="noopener noreferrer"
                                              target="_blank"
                                            >
                                              <a
                                                className="card-link"
                                                href="https://github.com/nkbt/react-copy-to-clipboard"
                                                rel="noopener noreferrer"
                                                target="_blank"
                                              >
                                                react-copy-to-cliboard
                                              </a>
                                            </CardLink>
                                          </CardLink>
                                           - for all my copying needs
                                        </li>
                                        <li>
                                          <CardLink
                                            href="https://highlightjs.org/"
                                            text="highlight.js"
                                          >
                                            <CardLink
                                              href="https://highlightjs.org/"
                                              rel="noopener noreferrer"
                                              target="_blank"
                                            >
                                              <a
                                                className="card-link"
                                                href="https://highlightjs.org/"
                                                rel="noopener noreferrer"
                                                target="_blank"
                                              >
                                                highlight.js
                                              </a>
                                            </CardLink>
                                          </CardLink>
                                           - code syntax highlighter
                                        </li>
                                      </ul>
                                      <CardSubtitle>
                                        <ForwardRef
                                          className="card-subtitle"
                                        >
                                          <div
                                            className="card-subtitle h6"
                                          >
                                            Build
                                          </div>
                                        </ForwardRef>
                                      </CardSubtitle>
                                      <ul>
                                        <li>
                                          <CardLink
                                            href="https://github.com/"
                                            text="Git"
                                          >
                                            <CardLink
                                              href="https://github.com/"
                                              rel="noopener noreferrer"
                                              target="_blank"
                                            >
                                              <a
                                                className="card-link"
                                                href="https://github.com/"
                                                rel="noopener noreferrer"
                                                target="_blank"
                                              >
                                                Git
                                              </a>
                                            </CardLink>
                                          </CardLink>
                                           - for the version control
                                        </li>
                                        <li>
                                          <CardLink
                                            href="https://travis-ci.org/"
                                            text="Travis"
                                          >
                                            <CardLink
                                              href="https://travis-ci.org/"
                                              rel="noopener noreferrer"
                                              target="_blank"
                                            >
                                              <a
                                                className="card-link"
                                                href="https://travis-ci.org/"
                                                rel="noopener noreferrer"
                                                target="_blank"
                                              >
                                                Travis
                                              </a>
                                            </CardLink>
                                          </CardLink>
                                           - for the building on commit
                                        </li>
                                        <li>
                                          <CardLink
                                            href="https://pages.github.com/"
                                            text="Github Pages"
                                          >
                                            <CardLink
                                              href="https://pages.github.com/"
                                              rel="noopener noreferrer"
                                              target="_blank"
                                            >
                                              <a
                                                className="card-link"
                                                href="https://pages.github.com/"
                                                rel="noopener noreferrer"
                                                target="_blank"
                                              >
                                                Github Pages
                                              </a>
                                            </CardLink>
                                          </CardLink>
                                           - for the deployment as a static web application
                                        </li>
                                      </ul>
                                      <CardSubtitle>
                                        <ForwardRef
                                          className="card-subtitle"
                                        >
                                          <div
                                            className="card-subtitle h6"
                                          >
                                            Test
                                          </div>
                                        </ForwardRef>
                                      </CardSubtitle>
                                      <ul>
                                        <li>
                                          <CardLink
                                            href="https://airbnb.io/enzyme/"
                                            text="Enzyme"
                                          >
                                            <CardLink
                                              href="https://airbnb.io/enzyme/"
                                              rel="noopener noreferrer"
                                              target="_blank"
                                            >
                                              <a
                                                className="card-link"
                                                href="https://airbnb.io/enzyme/"
                                                rel="noopener noreferrer"
                                                target="_blank"
                                              >
                                                Enzyme
                                              </a>
                                            </CardLink>
                                          </CardLink>
                                        </li>
                                      </ul>
                                    </div>
                                  </CardBody>
                                </WhatWasItMadeWithCardContent>
                              </div>
                            </ForwardRef>
                          </ListGroupItem>
                        </Bootstrap(ListGroupItem)>
                      </div>
                    </AbstractNav>
                  </ContextTransform(AbstractNav)>
                </ListGroup>
              </Bootstrap(ListGroup)>
            </Uncontrolled(Bootstrap(ListGroup))>
          </ForwardRef>
        </div>
      </Card>
    </div>
  </CardDeck>
</AboutMe>
`;
